src/PelicanFile.cc

```diff
@@ -17,23 +17,79 @@
  ***************************************************************/
 
 #include "FedInfo.hh"
+#include "ParseTimeout.hh"
 #include "PelicanFile.hh"
 #include "CurlOps.hh"
 #include "CurlUtil.hh"
 
+#include <XrdCl/XrdClConstants.hh>
+#include <XrdCl/XrdClDefaultEnv.hh>
 #include <XrdCl/XrdClLog.hh>
 #include <XrdCl/XrdClStatus.hh>
 #include <XrdCl/XrdClURL.hh>
 
 using namespace Pelican;
 
+struct timespec Pelican::File::m_min_client_timeout = {2, 0};
+struct timespec Pelican::File::m_default_header_timeout = {10, 0};
+
+struct timespec
+File::GetTimeoutFromHeader(const std::string &header_value, XrdCl::Log *logger)
+{
+    struct timespec ts = File::GetDefaultHeaderTimeout();
+    if (!header_value.empty()) {
+        std::string errmsg;
+        if (!ParseTimeout(header_value, ts, errmsg)) {
+            logger->Error(kLogXrdClPelican, "Failed to parse pelican.timeout parameter: %s", errmsg.c_str());
+        } else if (ts.tv_sec >= 1) {
+                ts.tv_sec--;
+        } else {
+            ts.tv_nsec /= 2;
+        }
```

Comment:
It's not quite halving -- it's subtracting one second from the timeout and, if the timeout is less than a second, _then_ it halves it.

The thinking is that if the client needs a response in N seconds, then we ought to set the internal timeout to (N-1) seconds to provide enough time for our response to arrive at the client.

Since it confused you, I'll add a comment along these lines.